### DISCUSSION TITLE
Another killer app for Dogecoin: Transaction lottery (PoTx) bridge to Defi Sidechain

### DISCUSSION THREAD

**Discussion by GiverofMemory** [dogecoin/dogecoin#2803] [General]
I gave this idea to the devs on bitcoin but I think it could be even more relevent for dogecoin since Billy Markus initially started doge from luckycoin and that had a lottery type of block reward, in fact I believe dogecoin block reward changed based on luck originally.

Here is the writeup I did on bitcoin talk https://bitcointalk.org/index.php?topic=5381931.msg59030237#msg59030237

Basically the idea is that if you get lucky on your transaction ID number, then you win tokens lets say.  In dogecoin, perhaps these tokens could be for the EVM sidechain (doge based defi chain).

The way it can be determined if a transaction is a winner could be something like hashing the TxID with the blockhash of the block it is included in, and also hashing that with the hash of the next block and so on.  Lets say we go 10 blocks deep.  Each transaction would be hashed with the future 10 blockhashes, and if that has has a certain number of leading zeroes, then the transaction is a winner.

How do they get rewarded?  The miner would have to look 10 blocks back, hash every txid with every blockhash of the last 10 blocks, and then if any of those hashes contains at least a certain number of leading zeroes, they get new defi doge token.  This token will be issued by the coinbase transaction that the miner specified in the block.  Then this defi doge token can be used to buy tokens or nft's on the dogecoin EVM sidechain.  This is a decentralized bridge from doge to defi doge.

In reality the coinbase transaction would send a spec of dogecoin dust (like 0.0000001 dogecoin or whatever) with a special op return tag that specifies the token.  And would probably make it so the token could only be transfered on the EVM sidechain.  Or the token could be tradable on dogecoin and then locked or burned to move it to the EVM sidechain.

**Comment by patricklodder** [comment]
Question: why issue a new token?

**Reply by tealethetoolman** [reply]
Because gas fees are great for everyone! /s

**Reply by patricklodder** [reply]
Does create opportunity: then someone can create againfees.wtf, and do an airdrop of yet another token. The possibilities are truly endless.

**Reply by GiverofMemory** [reply]
> Question: why issue a new token?

Good question.  So we have dogecoin.  Typically in a defi bridge like bitcoin cash's, the coin is locked on the main chain, and a wrapped version is released on the EVM sidechain.  The problem with this like Vitalik mentioned is that their is usually just a multisig wallet creating the wrapped coin.  It is basically similar to the "trust me bro" done by Tether releasing a new batch of tokens.  In the case of bitcoin cash (smartBCH) it is coinflex.com operating the bridge, which is the central authority sadly.

In this method I outline above, we finally solve how is "defi doge" created.  No central party has to create wrapped doge, we could create it by a transaction lottery totally separate from dogecoin mining itself.  In this way dogecoin would always be free from entanglement and never need to be locked or burned to create defi doge.  And random people will be the ones "mining" defi doge by just using dogecoin.

We could have it where you send defi doge (that was created in the transaction lottery) to a burn address on dogecoin and put your ETH address in the OP_RETURN and now you "created" the defi doge in a decentralized way and it is now in your ETH (EVM sidechain) wallet.  The sidechain would literally look at dogecoin and trace when 'defi doge' was created and the ETH address that was in the OP RETURN and would delegate the defi doge to that address.

Then for normal dogecoiners there would be swaps that will trade EVM defi doge for normal doge so you don't have to win the transaction lottery to get defi doge.

**Reply by tealethetoolman** [reply]
lets make this simple. Dogecoin is money. trying to dilute it with the introduction of the EVM is just going to muddy the waters and introduce more vectors for things to go wrong and for mining pools to take over the network causing fees to get out of hand. Could you image having to contribute 10k doge in fees to Vitaliks network every time you want to send 5 doge to someone just for the chance that maybe you'll get a token in a lottery which you'll need to spend another 10K doge just to unwrap it and make it into real doge.

**Reply by patricklodder** [reply]
I think this is a way to have a possibly-provable fair distribution scheme. That's cool.

I'm going to be a bit skeptical after that, please forgive.

I think I understand this as that we'd have no peg between this "defidoge" and DOGE. So this is then not much different from for example CLAM, that issued coin to a snapshot of BTC/LTC/DOGE holders. But... it's a completely new token/coin. It will trade against DOGE in AMMs/DEXs but maybe also tether on some centralized exchanges if it gets that far, like the original XDP did. So now you'll have people c&ping your code and launching their tokens, suckering people in to buy their token after they accumulated a lot of it, and we're back where we started. New token = new source of volatility = few will win, many will lose.

Jackson (Palmer) asked me last year: what if there was a blockchain without a token? It's an interesting thought experiment but for now let's limit the exploration to: how can we do sidechain without creating a new token? I have some ideas but I'm not going to advertise before PoC at the moment, because everything I say is pushed to twitter right now and used to manipulate the markets.

**Reply by GiverofMemory** [reply]
Yes, the goal being a provable fair distribution for an "official" EVM sidechain of doge, like SmartBCH is for bitcoin cash.

Yes it would be unpegged to dogecoin.  However one way we can tune the possible price to stay relatively pegged to doge is by tuning the difficulty of the "lottery".  So we know each transaction fee is 0.01 dogecoin.  So if we want to award 1 defidoge at a time, we could make it so on average 1 in 100 transactions is a winner.  The price would not be pegged, but the cost of producing a defidoge will be about 1 doge in transaction fees.  If the minimum fee is ever changed, so could the difficulty requirement of the lucky transaction.  It's honestly a lot like mining, where the cost to mine a bitcoin in electricity is basically the floor price for bitcoin.  In our case the "mining" would be just making a transaction on dogecoin.

I like the idea of a blockchain/sidechain without a token and will think about it a bit.  One design I do have is sort of like a lightning network without a token, it is based on IOU's to people you trust, but I will keep thinking about how to do a sidechain without a token. https://web.archive.org/web/20210414170746/https://www.naturehacker.org/2017/10/the-only-100-fully-decentralized-peer.html

**Reply by patricklodder** [reply]
> In our case the "mining" would be just making a transaction on dogecoin.

I've seen some of this on EOS where it ultimately caused bots to spam the blockchain to oblivion. This happened on 2 occasions. Because what if I as a whale pump smartdoge and suddenly it's worth 100,000 DOGE each (because I control all the circulating supply, I set the price). Then suddenly it's economical to make full blocks until 25 DOGE/tx fees just by sending money between my change addresses and paying the fees.

**Reply by GiverofMemory** [reply]
> > In our case the "mining" would be just making a transaction on dogecoin.
> 
> I've seen some of this on EOS where it ultimately caused bots to spam the blockchain to oblivion. This happened on 2 occasions. Because what if I as a whale pump smartdoge and suddenly it's worth 100,000 DOGE each (because I control all the circulating supply, I set the price). Then suddenly it's economical to make full blocks until 25 DOGE/tx fees just by sending money between my change addresses and paying the fees.

Yes that can be a possibility.  One thing that would mitigate this would be something I wrote in https://github.com/dogecoin/dogecoin/discussions/2809 which would be a combination of this idea and also a fork giving defidoge to all dogecoin holders 1:1.  So that fork would be the initial distribution (so one whale couldn't corner the supply) then future coins would be created by this transaction lottery idea.  But yes with this idea it may cause an increase in transactions in any case.  But I think with an initial distribution like mentioned above, it would cause much less fomo transaction spamming.  Also if we see spam, we can always increase the rarity of winning the transaction lottery, making it less financially viable to spam.  The same sort of mentality is used in creating the min tx fee.
